File: mobile/src/types/compare.ts
export interface MarketResult {
  supermarket: string;
  product_name: string;
  price: number;
  price_per_unit: string | null;
  url: string;
  image_url: string | null;
}

export interface SupermarketGroup {
  supermarket: string;
  products: MarketResult[];
}

export interface CompareResponse {
  query: string;
  cheapest: MarketResult | null;
  by_supermarket: SupermarketGroup[];
  from_cache: boolean;
  warnings: string[];
}

File: backend/api/schemas/compare.py
from pydantic import BaseModel, Field


class MarketResult(BaseModel):
    supermarket: str
    product_name: str
    price: float
    price_per_unit: str | None = None
    url: str
    image_url: str | None = None


class SupermarketGroup(BaseModel):
    supermarket: str
    products: list[MarketResult]


class CompareResponse(BaseModel):
    query: str
    cheapest: MarketResult | None
    by_supermarket: list[SupermarketGroup]
    from_cache: bool = False
    warnings: list[str] = Field(default_factory=list)
